tests/pipelines/integration.nf.test

```diff
@@ -0,0 +1,244 @@
+            assert iridanext_metadata.GHI.metadata_1 == "3.1"
+            assert iridanext_metadata.GHI.metadata_2 == "3.2"
+            assert iridanext_metadata.GHI.metadata_3 == "3.3"
+            assert iridanext_metadata.GHI.metadata_4 == "3.4"
+            assert iridanext_metadata.GHI.metadata_5 == "3.5"
+            assert iridanext_metadata.GHI.metadata_6 == "3.6"
+            assert iridanext_metadata.GHI.metadata_7 == "3.7"
+            assert iridanext_metadata.GHI.metadata_8 == "3.8"
+            */
+        }
+    }
+
+    test("Invalid transformation") {
+        tag "pipeline_invalid_transformation"
+
+        when {
+            params {
+                input = "$baseDir/tests/data/samplesheets/basic.csv"
+                outdir = "results"
+
+                transformation = "UNKNOWN123"
+            }
+        }
+
+        then {
+            assert workflow.failed
+            def output = workflow.stdout.join("\n")
+            assert output.contains("Validation of pipeline parameters failed!")
+        }
+    }
+
+    test("Null transformation") {
+        tag "pipeline_null_transformation"
+
+        when {
+            params {
+                input = "$baseDir/tests/data/samplesheets/basic.csv"
+                outdir = "results"
+
+                transformation = null
+            }
+        }
+
+        then {
+            assert workflow.failed
+            def output = workflow.stdout.join("\n")
+            assert output.contains("Unspecified transformation '--transformation'. Exiting now.")
+        }
+    }
+
+    test("Basic lock - missing metadata") {
+        tag "pipeline_missing_metadata"
+
+        when {
+            params {
+                input = "$baseDir/tests/data/samplesheets/missing_metadata.csv"
+                outdir = "results"
+
+                transformation = "lock"
+            }
+        }
+
+        then {
+            assert workflow.success
+            assert path("$launchDir/results").exists()
+
+            // Check Locked Results
+            def locked = path("$launchDir/results/transformation/transformation.csv")
+            assert locked.exists()
+
+            assert locked.text.contains("sample_id,irida_id,metadata_1,metadata_2,metadata_3,metadata_4,metadata_5,metadata_6,metadata_7,metadata_8")
+            assert locked.text.contains("ABC,sample1,,1.2,1.3,,1.5,1.6,,1.8")
+            assert locked.text.contains("sample2,sample2,2.1,,2.3,2.4,,2.6,2.7,")
+            assert locked.text.contains("GHI,sample3,3.1,3.2,,3.4,3.5,,3.7,3.8")
+
+            // Check IRIDA Next JSON Output
+            def iridanext_json = path("$launchDir/results/iridanext.output.json").json
+            def iridanext_metadata = iridanext_json.metadata.samples
+
+            assert iridanext_metadata.size() == 3
+
+            assert iridanext_metadata.containsKey("ABC")
+            assert iridanext_metadata.ABC.irida_id == "sample1"
```

Comment:
I don't think we want to write `irida_id` to the metadata table so this key should be put into the `ignore` list of keys in the `iridanext.config` for metadata.

Reply:
Corrected in 2de1af10c7dd6b357fee28c5607e344c96c0b02a